src/node_worker.cc

```diff
@@ -644,7 +644,37 @@ void Worker::StartThread(const FunctionCallbackInfo<Value>& args) {
           w->JoinThread();
           // implicitly delete w
         });
-  }, static_cast<void*>(w)), 0);
+  }, static_cast<void*>(w));
+  if (ret != 0) {
+    char err_buf[128];
+    uv_err_name_r(ret, err_buf, sizeof(err_buf));
+    w->custom_error_ = "ERR_WORKER_INIT_FAILED";
+    w->custom_error_str_ = err_buf;
+    w->loop_init_failed_ = true;
+    w->thread_joined_ = true;
+    w->stopped_ = true;
+    w->env()->remove_sub_worker_context(w);
+    {
+      HandleScope handle_scope(w->env()->isolate());
+      Context::Scope context_scope(w->env()->context());
+      // Reset the parent port as we're closing it now anyway.
+      w->object()->Set(w->env()->context(),
+                    w->env()->message_port_string(),
+                    Undefined(w->env()->isolate())).Check();
+      Local<Value> args[] = {
+          Integer::New(w->env()->isolate(), w->exit_code_),
+          w->custom_error_ != nullptr
+              ? OneByteString(w->env()->isolate(), w->custom_error_).As<Value>()
+              : Null(w->env()->isolate()).As<Value>(),
+          !w->custom_error_str_.empty()
+              ? OneByteString(w->env()->isolate(), w->custom_error_str_.c_str())
+                    .As<Value>()
+              : Null(w->env()->isolate()).As<Value>(),
+      };
```

Comment:
I find this ternary-heave style very hard to read. Can I suggest this?
```c++
Isolate* isolate = w->env()->isolate();
HandleScope handle_scope(isolate);
// ...
Local<Value> exit_code_v = Integer::New(isolate, w->exit_code_);
Local<Value> null_v = Null(isolate).As<Value>();
Local<Value> argv[3] = {exit_code_v, null_v, null_v};

if (w->custom_error_ != nullptr)
  args[1] = OneByteString(isolate, w->custom_error_);

if (!w->custom_error_str_.empty())
  args[2] = OneByteString(isolate, w->custom_error_str_.c_str());
```
Or alternatively:
```c++
Local<Value> exit_code_v = Integer::New(isolate, w->exit_code_);
Local<Value> null_v = Null(isolate).As<Value>();
Local<Value> custom_error_v = null_v;
Local<Value> custom_error_str_v = null_v;

if (w->custom_error_ != nullptr)
  custom_error_v = OneByteString(isolate, w->custom_error_);

if (!w->custom_error_str_.empty())
  custom_error_str_v = OneByteString(isolate, w->custom_error_str_.c_str());

Local<Value> argv[] = {exit_code_v, custom_error_v, custom_error_str_v};
```
I've left out the `.As<Value>()` casts because I don't think they're necessary here.

Reply:
Fwiw, we don’t need the ternaries here anyway because we already know that `custom_error_` and `custom_error_str_` are set.

But either way, that should go away if this is converted to throwing an exception.